core/src/main/java/org/elasticsearch/index/query/QueryBuilders.java

```diff
@@ -844,6 +850,65 @@ public static PercolatorQueryBuilder percolatorQuery(String documentType, String
                 indexedDocumentRouting, indexedDocumentPreference, indexedDocumentVersion);
     }
 
+    public static void registerBuiltinBuilders(SearchModule searchModule, NamedWriteableRegistry namedWriteableRegistry) {
```

Comment:
Have a look at what I proposed for ScoreFunctionParsers this morning. It is one extra class but I think I can make up for it by allowing us to remove all the parser classes. It is one extra class but not one extra per query.